plugins/modules/nxos_fc_interfaces.py

```diff
@@ -0,0 +1,491 @@
+#   trunk_mode: off
+#   enabled: True
+# commands:
+# - interface fc1/2
+# - no switchport speed 1000
+# - no switchport mode E
+# - switchport trunk mode on
+# - shutdown
+# after:
+# - name: fc1/2
+#   speed: auto
+#   mode: auto
+#   trunk_mode: on
+#   enabled: False
+
+# After state:
+# ------------
+#
+# switch# show running-config interface all
+# interface fc1/2
+#     switchport speed auto
+#     switchport mode auto
+#     no switchport description
+#     switchport trunk mode on
+#     shutdown
+
+# Using rendered
+
+- name: Use rendered state to convert task input to device specific commands
+  cisco.nxos.nxos_fc_interfaces:
+    config:
+    - name: fc1/1
+      speed: auto
+      mode: auto
+      trunk_mode: on
+      enabled: True
+      description: This is a sample line
+    - name: fc1/2
+      speed: 1000
+      mode: E
+      trunk_mode: off
+      enabled: True
+      state: rendered
+
+# Task Output
+# -----------
+#
+# rendered:
+# interface fc1/1
+#     switchport speed auto
+#     switchport mode auto
+#     switchport description This is a sample line
+#     switchport trunk mode on
+#     no shutdown
+#
+# interface fc1/2
+#     switchport speed 1000
+#     switchport mode E
+#     no switchport description
+#     switchport trunk mode off
+#     no shutdown
+
+# Using parsed
+
+# parsed.cfg
+# ------------
+#
+# interface fc1/1
+#     switchport speed auto
+#     switchport mode auto
+#     switchport description This is a sample line
+#     switchport trunk mode on
+#     no shutdown
+#
+# interface fc1/2
+#     switchport speed 1000
+#     switchport mode E
+#     no switchport description
+#     switchport trunk mode off
+#     no shutdown
+
+- name: Use parsed state to convert externally supplied config to structured format
+  cisco.nxos.nxos_fc_interfaces:
+    running_config: "{{ lookup('file', 'parsed.cfg') }}"
+    state: parsed
+
+# Task output
+# -----------
+#
+#  parsed:
+# - name: fc1/1
+#   speed: auto
+#   mode: auto
+#   trunk_mode: on
+#   enabled: True
+#   description: This is a sample line
+# - name: fc1/2
+#   speed: 1000
+#   mode: E
+#   trunk_mode: off
+#   enabled: True
+
+# Using gathered
+
+# Before state:
+# -------------
+#
+# switch# show running-config | section interface
+# interface fc1/1
+#     switchport speed auto
+#     switchport mode auto
+#     switchport description This is a sample line
+#     switchport trunk mode on
+#     no shutdown
+#
+# interface fc1/2
+#     switchport speed 1000
+#     switchport mode E
+#     no switchport description
+#     switchport trunk mode off
+#     no shutdown
+#
+- name: Gather interfaces facts from the device using nxos_fc_interfaces
+  cisco.nxos.nxos_fc_interfaces:
+    state: gathered
+#
+# Task output
+# -----------
+#
+# - name: fc1/1
+#   speed: auto
+#   mode: auto
+#   trunk_mode: on
+#   enabled: True
+#   description: This is a sample line
+# - name: fc1/2
+#   speed: 1000
+#   mode: E
+#   trunk_mode: off
+#   enabled: True
+
```

Comment:
Can you please add an example for `overridden`?

Reply:
overrideen and replaced are same what will be the difference?

Reply:
I'm not sure I get it. Why is overridden and replaced state the same?

Reply:
i think we need a call to over these things

Reply:
Sure, please feel free to send an invite.